### PR TITLE
fixed There may be concurrency problems in org.redisson.RedissonLock#tryLockInnerAsync issue #3484

### PR DESCRIPTION
**Pull request by coding-tortoise**
issue #3484 
Signed-off-by: 小乌龟不会写代码 <41791762+coding-tortoise@users.noreply.github.com>

### CODE REVIEW

**Comment by coding-tortoise** [comment]
set internalLockLeaseTime only when the lock acquired

File: redisson/src/main/java/org/redisson/RedissonLock.java [hunk 1]
```diff
@@ -144,38 +144,51 @@ private Long tryAcquire(long waitTime, long leaseTime, TimeUnit unit, long threa
     }
     
     private RFuture<Boolean> tryAcquireOnceAsync(long waitTime, long leaseTime, TimeUnit unit, long threadId) {
-        if (leaseTime != -1) {
-            return tryLockInnerAsync(waitTime, leaseTime, unit, threadId, RedisCommands.EVAL_NULL_BOOLEAN);
+        RFuture<Boolean> ttlRemainingFuture;
+        if (-1 != leaseTime) {
```
Comment (mrniko) [inline redisson/src/main/java/org/redisson/RedissonLock.java:148]: rewrite it as `leaseTime = -1`

File: redisson/src/main/java/org/redisson/RedissonLock.java [hunk 2]
```diff
@@ -144,38 +144,51 @@ private Long tryAcquire(long waitTime, long leaseTime, TimeUnit unit, long threa
...
+
         ttlRemainingFuture.onComplete((ttlRemaining, e) -> {
             if (e != null) {
                 return;
             }
 
             // lock acquired
             if (ttlRemaining) {
-                scheduleExpirationRenewal(threadId);
+                if (-1 != leaseTime) {
```
Comment (mrniko) [inline redisson/src/main/java/org/redisson/RedissonLock.java:162]: rewrite it as `leaseTime = -1`

File: redisson/src/main/java/org/redisson/RedissonLock.java [hunk 3]
```diff
@@ -144,38 +144,51 @@ private Long tryAcquire(long waitTime, long leaseTime, TimeUnit unit, long threa
...
-                                                                TimeUnit.MILLISECONDS, threadId, RedisCommands.EVAL_LONG);
         ttlRemainingFuture.onComplete((ttlRemaining, e) -> {
             if (e != null) {
                 return;
             }
 
             // lock acquired
             if (ttlRemaining == null) {
-                scheduleExpirationRenewal(threadId);
+                if (-1 != leaseTime) {
```
Comment (mrniko) [inline redisson/src/main/java/org/redisson/RedissonLock.java:187]: rewrite it as `leaseTime = -1`

**Review by mrniko — CHANGES_REQUESTED** [review]

**Review by mrniko — APPROVED** [review]

**Comment by mrniko** [comment]
Thanks for contribution!

**Pull request merged**